applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/types.h>
+#include <zephyr/bluetooth/conn.h>
+
+#include <caf/events/ble_common_event.h>
+#include <caf/events/ble_smp_event.h>
+#include "config_event.h"
+#include "usb_event.h"
+#include <caf/events/button_event.h>
+#include <caf/events/power_event.h>
+
+#define MODULE dvfs
+#include <caf/events/module_state_event.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_DVFS_LOG_LEVEL);
+
+#include "ld_dvfs_handler.h"
+
+#define DVFS_INIT_TIMEOUT		K_SECONDS(1)
+#define REG_CONN_INTERVAL_LLPM_MASK	0x0d00
+
+enum dvfs_states_list {
```

Comment:
Naming here is a bit misleading - these are not mutually exclusive states.

Maybe we can e.g. make it `app_state` call members like: `APP_STATE_USB_CONNECTED`, `APP_LLPM_CONNECTED`? We also do not need the `DEFAULT` element here (default is used when no other bit is set). You could add `APP_STATE_INITIALIZING` and set in on boot to ensure we use high frequency until app is fully initialized.

Reply:
Discussed offline, we will use `dvfs_state` and call members `DVFS_STATE_INITIALIZING`, `DVFS_STATE_USB_CONNECTED`, `DVFS_STATE_USB_CONNECTED`, `DVFS_STATE_LLPM_CONNECTED`

Reply:
disused offline will go with dvfs_state